### ISSUE TITLE
v1.9.1.4 exception spam - no longer mining

### ISSUE THREAD

**Issue by F1Rumors** [nicehash/NiceHashMiner#1551]
The miner was running, but is now reporting exceptions almost continuously, and failing to start the miners.  The exceptions relate to failing to get power information (apparently), though there may be something else going on in there.

Did not catch the genesis - will try again later and attempt to see what caused this.  Log sample follows:

```cs
[2019-06-04 13:19:05] [ERROR] [NVML] System.Exception: NVML get temp failed with code: Unknown
   at NiceHashMiner.Devices.CudaComputeDevice.get_Temp() in D:\Programming\NiceHashMinerLegacy\src\NiceHashMinerLegacy\Devices\ComputeDevice\CudaComputeDevice.cs:line 66
[2019-06-04 13:19:05] [ERROR] [NVML] System.Exception: NVML get load failed with code: Unknown
   at NiceHashMiner.Devices.CudaComputeDevice.get_Load() in D:\Programming\NiceHashMinerLegacy\src\NiceHashMinerLegacy\Devices\ComputeDevice\CudaComputeDevice.cs:line 41
[2019-06-04 13:19:05] [ERROR] [NVML] System.Exception: NVML get temp failed with code: Unknown
   at NiceHashMiner.Devices.CudaComputeDevice.get_Temp() in D:\Programming\NiceHashMinerLegacy\src\NiceHashMinerLegacy\Devices\ComputeDevice\CudaComputeDevice.cs:line 66
[2019-06-04 13:19:05] [ERROR] [NVML] System.Exception: NVML get load failed with code: Unknown
   at NiceHashMiner.Devices.CudaComputeDevice.get_Load() in D:\Programming\NiceHashMinerLegacy\src\NiceHashMinerLegacy\Devices\ComputeDevice\CudaComputeDevice.cs:line 41
[2019-06-04 13:19:05] [ERROR] [NVML] System.Exception: NVML get temp failed with code: Unknown
   at NiceHashMiner.Devices.CudaComputeDevice.get_Temp() in D:\Programming\NiceHashMinerLegacy\src\NiceHashMinerLegacy\Devices\ComputeDevice\CudaComputeDevice.cs:line 66
[2019-06-04 13:19:05] [ERROR] [NVML] System.Exception: NVML get load failed with code: Unknown
   at NiceHashMiner.Devices.CudaComputeDevice.get_Load() in D:\Programming\NiceHashMinerLegacy\src\NiceHashMinerLegacy\Devices\ComputeDevice\CudaComputeDevice.cs:line 41
[2019-06-04 13:19:05] [ERROR] [Miner149-GMiner-algo(Beam)-devs(NVIDIA:2)] Error occured while getting API stats: An error occurred while sending the request.
[2019-06-04 13:19:05] [ERROR] [NVML] System.Exception: NVML power get failed with status: Unknown
   at NiceHashMiner.Devices.CudaComputeDevice.get_PowerUsage() in D:\Programming\NiceHashMinerLegacy\src\NiceHashMinerLegacy\Devices\ComputeDevice\CudaComputeDevice.cs:line 107
[2019-06-04 13:19:05] [ERROR] [NVML] System.Exception: NVML power get failed with status: Unknown
   at NiceHashMiner.Devices.CudaComputeDevice.get_PowerUsage() in D:\Programming\NiceHashMinerLegacy\src\NiceHashMinerLegacy\Devices\ComputeDevice\CudaComputeDevice.cs:line 107
[2019-06-04 13:19:05] [INFO] [Miner150-NBMiner-algo(GrinCuckatoo31)-devs(NVIDIA:0,NVIDIA:1)] Starting miner commandLine='-a cuckatoo -o stratum+tcp://grincuckatoo31.usa.nicehash.com:3372 -u 35MQaSQkNoprbFX9yQJsUu3gkk3KAru6qU.worker1 --api 127.0.0.1:4000 -d 0,1 -RUN '
[2019-06-04 13:19:05] [INFO] [Miner150-NBMiner-algo(GrinCuckatoo31)-devs(NVIDIA:0,NVIDIA:1)] Starting miner environmentVariables='<null>'
[2019-06-04 13:19:05] [INFO] [Miner149-GMiner-algo(Beam)-devs(NVIDIA:2)] Starting miner commandLine='-a 150_5 -s beam.usa.nicehash.com -n 3370 -u 35MQaSQkNoprbFX9yQJsUu3gkk3KAru6qU.worker1 -d 2 -w 0 --api 4007 '
[2019-06-04 13:19:05] [INFO] [Miner149-GMiner-algo(Beam)-devs(NVIDIA:2)] Starting miner environmentVariables='<null>'
[2019-06-04 13:19:06] [ERROR] [NVML] System.Exception: NVML get load failed with code: Unknown
   at NiceHashMiner.Devices.CudaComputeDevice.get_Load() in D:\Programming\NiceHashMinerLegacy\src\NiceHashMinerLegacy\Devices\ComputeDevice\CudaComputeDevice.cs:line 41
[2019-06-04 13:19:06] [ERROR] [NVML] System.Exception: NVML get temp failed with code: Unknown
   at NiceHashMiner.Devices.CudaComputeDevice.get_Temp() in D:\Programming\NiceHashMinerLegacy\src\NiceHashMinerLegacy\Devices\ComputeDevice\CudaComputeDevice.cs:line 66
[2019-06-04 13:19:06] [ERROR] [Miner150-NBMiner-algo(GrinCuckatoo31)-devs(NVIDIA:0,NVIDIA:1)] Error occured while getting API stats: An error occurred while sending the request.
[2019-06-04 13:19:06] [ERROR] [NVML] System.Exception: NVML power get failed with status: Unknown
```

**Comment by S74nk0** [comment]
@F1Rumors We are working on a re-write for device detection and device monitoring. This re-write will introduce a fix to this log spam and device monitoring recovery. The genesis of this is NVIDIA driver crash probably caused by some miner.

**Comment by F1Rumors** [comment]
@S74nk0 thank you, that makes sense.  I think the ClaymoreDual miner does not play nicely on my setup -- at first glance that appears to be the common factor for crashing the NVIDIA driver, I can investigate from there for that aspect.

**Comment by S74nk0** [comment]
New version released [1.9.1.5](https://github.com/nicehash/NiceHashMinerLegacy/releases/tag/1.9.1.5) closing this issue. If the issue is present in the new release please open a new issue.

Closing.
